File: codeChef/easy/onp.py
from sys import stdin

def parse(expr, i):
  i += 1

  if expr[i] == '(':
    op1, i = parse(expr, i)
  else:
    op1 = expr[i]
    i += 1
	
  opt = expr[i]
  i += 1

  if expr[i] == '(':
    op2, i = parse(expr, i)
  else:	
    op2 = expr[i]
    i += 1

  i += 1
return op1 + op2 + opt, i 


stdin.readline()
for expr in stdin:
  rpn, i = parse(expr, 0)
  print rpn

File: spoj/onp.rb
def parse(s, i)
  i += 1
  op1, i = s[i] == '(' ? parse(s, i) : [s[i], i+1]
  opt, i = s[i], i+1
  op2, i = s[i] == '(' ? parse(s, i) : [s[i], i+1]
  [op1 + op2 + opt, i + 1]
end

gets.to_i.times do |_|
  res, _ = parse(gets, 0)
  puts res
end
